src/samba.yml

```diff
@@ -0,0 +1,51 @@
+---
+- hosts: all
+  name: Install Samba server
+  become: yes
+  become_method: sudo
+  roles:
+    - role: samba
+      vars:
+        create_guest_user: yes
+        server: yes
+  tasks:
+    # This way samba should not start until after the EFS share is
+    # mounted.
+    - name: Configure Samba unit file
+      block:
+        - name: Make Samba start after /share is mounted
+          community.general.ini_file:
+            create: no
+            # SystemD unit files don't use extra spaces
+            no_extra_spaces: yes
+            option: /share
+            path: /lib/systemd/system/smbd.service
+            section: Unit
+            value: RequiresMountsFor
+    - name: Configure Samba
+      block:
+        - name: Send logs to syslog as well
+          # The ini_file module won't handle tacking extra text onto a
+          # line, so we have to use replace for this.
+          ansible.builtin.replace:
+            path: /etc/samba/smb.conf
+            # The negative lookahead makes this regex match lines that
+            # start with "logging=" but _do not_ contain "syslog@1".
+            regexp: ^(\s*logging=.*(?!syslog@1).*)
+            replace: \1 syslog@1
+            validate: /usr/bin/testparm --suppress-prompt %s
+        - name: Share EFS mount
+          community.general.ini_file:
+            create: no
+            option: "{{ item.option }}"
+            path: /etc/samba/smb.conf
+            section: Share
+            value: "{{ item.value }}"
+          loop:
+            - {option: "browseable", value: "yes"}
+            - {option: "guest ok", value: "yes"}
+            - {option: "guest only", value: "yes"}
+            - {option: "map to guest", value: "Bad User"}
```

Comment:
I'm not sure this warrants a comment, but this seemed like a magic string to me.  Found my answer in the `smb.conf` man page: 

```
This parameter can take four different values, which tell smbd(8) what to do with user login requests that 
don't match a valid UNIX user in some way... 

Bad User - Means user logins with an invalid password are rejected, unless the username does not exist,
in which case it is treated as a guest login and mapped into the guest account.
```

Reply:
In my opinion, that's interesting enough for a comment.

Reply:
Please see commit db35b07.